source/client.c

```diff
@@ -568,6 +568,10 @@ int aws_mqtt_client_connection_set_will(
     bool retain,
     const struct aws_byte_cursor *payload) {
 
+    AWS_PRECONDITION(connection);
+    AWS_PRECONDITION(topic);
+
+    bool error = true;
```

Comment:
rather than error, just make it "int result = AWS_OP_ERR;" and return it at the end